### PR TITLE
Test to automate validation of performance profiles and change the performance profiles post deployment

### PR DESCRIPTION
**Pull request by pintojoy**
This automates test cases from [RHSTOR-4547](https://issues.redhat.com//browse/RHSTOR-4547)

### CODE REVIEW

**Comment by ramkiperiy** [comment]
Is this a common test to validate any of the profile change?, from TC name it looks like its common test or is it specific to one particular profile.

**Comment by pintojoy** [comment]
> @pintojoy Is this a common test to validate any of the profile change?, from TC name it looks like its common test or is it specific to one particular profile.

@ramkiperiy Its a common test.. if the selected profile is differnet than existing profile, profile change takes place

File: tests/functional/z_cluster/test_performance_profile_validation.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+log = logging.getLogger(__name__)
+
+
+@brown_squad
+@tier4a
+@skipif_external_mode
+@skipif_managed_service
+@skipif_ocs_version("<4.15")
+@pytest.mark.polarion_id("XXXX")
```
Comment (ramkiperiy) [inline tests/functional/z_cluster/test_performance_profile_validation.py:34]: Planing to create TC in polrion or waiting for polarion data move?

**Comment by pintojoy** [comment]
Both the test are verified via PR validation jobs. Adding verified marker

File: tests/functional/z_cluster/test_performance_profile_validation.py (hunk 1 above)
Comment (pintojoy) [reply tests/functional/z_cluster/test_performance_profile_validation.py:34]: Yes @ramkiperiy

**Comment by ramkiperiy** [comment]
@pintojoy you need to resolve python 3.9 related checks, for the PR to get merged

File: tests/functional/z_cluster/test_performance_profile_validation.py (hunk 1 above)
Comment (ramkiperiy) [reply tests/functional/z_cluster/test_performance_profile_validation.py:34]: Do add the polarion ID, its generated and updated in test plan.

File: tests/functional/z_cluster/test_performance_profile_validation.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                ), f"Resource values arent reflecting actual values for {label} pod "
+        log.info("All the memory and CPU values are matching in the profile")
+
+    @pytest.mark.parametrize(
+        argnames=["perf_profile"],
+        argvalues=[
+            pytest.param(*["performance"], marks=pytest.mark.polarion_id("OCS-XXXX")),
+            pytest.param(*["lean"], marks=pytest.mark.polarion_id("OCS-XXXX")),
+            pytest.param(*["balanced"], marks=pytest.mark.polarion_id("OCS-XXXX")),
+        ],
```
Comment (ramkiperiy) [inline tests/functional/z_cluster/test_performance_profile_validation.py:112]: Add polarion ID for the tests

**Review by ypersky1980 — APPROVED** [review]
LGTM

**Review by ramkiperiy — APPROVED** [review]

File: tests/functional/z_cluster/test_performance_profile_validation.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        else:
+            log.error("Does not match any performance profiles")
+
+        log.info(performance_profile)
+
+        label_selector = list(expected_cpu_limit_values.keys())
+
+        for label in label_selector:
+            for pod in get_pods_having_label(
+                label=label, namespace=config.ENV_DATA["cluster_namespace"]
```
Comment (yitzhak12) [inline tests/functional/z_cluster/test_performance_profile_validation.py:89]: The logic does not seem to work correctly. The `label_selector` is the pod values: 'mgr', mon', 'osd', etc.
But the `label` values are different, as you can see here: https://github.com/pintojoy/ocs-ci/blob/ca45897ca2d0e41f1f51347950d45951fea79caf/ocs_ci/ocs/constants.py#L381. For example: `app=rook-ceph-mon`.

Comment (yitzhak12) [reply tests/functional/z_cluster/test_performance_profile_validation.py:89]: Actually, I checked it in my local, and this will also work.

**Review by yitzhak12 — APPROVED** [review]

**Pull request merged**